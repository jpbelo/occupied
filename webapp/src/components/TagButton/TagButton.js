import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import {
  applyStyleModifiers,
  styleModifierPropTypes,
} from 'styled-components-modifiers'

const MODIFIER_CONFIG = {
  primary: ({ theme }) => css`
    background-color: ${theme.colors.grey.light};
    &[href^='http'] {
      background-color: ${theme.colors.primary.bgAlt};
      &:hover {
        background: ${theme.colors.primary.bg};
      }
    }
  `,
  secondary: ({ theme }) => css`
    background-color: ${theme.colors.grey.light};
    &[href^='http'] {
      background-color: ${theme.colors.secondary.bgAlt};
      &:hover {
        background: ${theme.colors.secondary.bg};
      }
    }
  `,
  asButton: () => css`
    font-size: 11px;
    display: inline-block;
    border-radius: 6px;
    padding: 0 10px;
    margin-right: 6px;
    margin-bottom: 6px;
    line-height: 2;
    transition: background-color 250ms ease-in-out;
  `,
  asText: ({ theme }) => css`
    font-size: 11px;
    color: ${theme.colors.grey.bgAlt};
    &:not(:last-child)::after {
      content: ', ';
    }
    &[href^='http'] {
      color: ${theme.colors.grey.bg};
      &:hover {
        color: ${theme.colors.primary.main};
      }
    }
  `,
}

const StyledTagButton = styled.a`
  ${applyStyleModifiers(MODIFIER_CONFIG)};
`

export default class TagButton extends Component {
  render() {
    const { url, name, modifiers } = this.props
    const layoutModifiers = modifiers || ['asButton', 'primary']

    return (
      <StyledTagButton
        as={!url && 'span'}
        target={url && '_blank'}
        href={url}
        modifiers={layoutModifiers}
      >
        {name}
      </StyledTagButton>
    )
  }
}

TagButton.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string,
  modifiers: styleModifierPropTypes(MODIFIER_CONFIG),
}
