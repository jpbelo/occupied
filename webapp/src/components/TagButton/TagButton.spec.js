import React from 'react'
import 'jest-styled-components'

import { renderWithTheme } from 'helpers/renderWithTheme'
import TagButton from './TagButton'

const mockProps = {
  name: 'Google',
  url: 'http://google.com',
}
const mockPropsNoUrl = {
  name: 'Google',
}

test('Render as Button', () => {
  let tree = renderWithTheme(<TagButton {...mockProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Render as Button without url', () => {
  let tree = renderWithTheme(<TagButton {...mockPropsNoUrl} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Render as Text', () => {
  let tree = renderWithTheme(
    <TagButton {...mockProps} modifiers="asText" />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Render as Text without url', () => {
  let tree = renderWithTheme(
    <TagButton {...mockPropsNoUrl} modifiers="asText" />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
