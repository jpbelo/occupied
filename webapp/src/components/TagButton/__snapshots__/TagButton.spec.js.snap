// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render as Button 1`] = `
.c0 {
  font-size: 11px;
  display: inline-block;
  border-radius: 6px;
  padding: 0 10px;
  margin-right: 6px;
  margin-bottom: 6px;
  line-height: 2;
  -webkit-transition: background-color 250ms ease-in-out;
  transition: background-color 250ms ease-in-out;
  background-color: hsl(0,0%,90%);
}

.c0[href^='http'] {
  background-color: hsl(0,50%,90%);
}

.c0[href^='http']:hover {
  background: hsl(0,50%,75%);
}

<a
  className="c0"
  href="http://google.com"
  target="_blank"
>
  Google
</a>
`;

exports[`Render as Button without url 1`] = `
.c0 {
  font-size: 11px;
  display: inline-block;
  border-radius: 6px;
  padding: 0 10px;
  margin-right: 6px;
  margin-bottom: 6px;
  line-height: 2;
  -webkit-transition: background-color 250ms ease-in-out;
  transition: background-color 250ms ease-in-out;
  background-color: hsl(0,0%,90%);
}

.c0[href^='http'] {
  background-color: hsl(0,50%,90%);
}

.c0[href^='http']:hover {
  background: hsl(0,50%,75%);
}

<span
  className="c0"
>
  Google
</span>
`;

exports[`Render as Text 1`] = `
.c0 {
  font-size: 11px;
  color: hsl(0,0%,50%);
}

.c0:not(:last-child)::after {
  content: ', ';
}

.c0[href^='http'] {
  color: hsl(0,0%,40%);
}

.c0[href^='http']:hover {
  color: hsl(0,97%,61%);
}

<a
  className="c0"
  href="http://google.com"
  target="_blank"
>
  Google
</a>
`;

exports[`Render as Text without url 1`] = `
.c0 {
  font-size: 11px;
  color: hsl(0,0%,50%);
}

.c0:not(:last-child)::after {
  content: ', ';
}

.c0[href^='http'] {
  color: hsl(0,0%,40%);
}

.c0[href^='http']:hover {
  color: hsl(0,97%,61%);
}

<span
  className="c0"
>
  Google
</span>
`;
